app/controllers/course/object_duplications_controller.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+class Course::ObjectDuplicationsController < Course::ComponentController
+  before_action :authorize_duplication
+
+  def new
+    load_target_courses_data
+    load_assessments_component_data
+  end
+
+  def create
+    job = Course::ObjectDuplicationJob.perform_later(
+      current_course, authorized_target_course, objects_to_duplicate, current_user: current_user
+    ).job
+    render json: { redirect_url: job_path(job) }
+  end
+
+  protected
+
+  def authorize_duplication
+    authorize!(:duplicate, current_course)
+  end
+
+  private
+
+  def load_target_courses_data
```

Comment:
I'm thinking if we should organise this into separate concerns, so each component can have 1 separate set of concerns, then declare this method in a `before_action` callback. Thoughts?

Reply:
I think the data loading belongs in the controller and is very specific to the current page. Perhaps what we could do to tidy up is to factor out some scopes if they are general enough. Will also use a single method to load data for components instead of one for each component.

Reply:
Yup you are right, the inclusion of each component in this page is deliberate and explicit. 👍